### ISSUE TITLE
no-extra-parens when used to disambiguate boolean operators

### ISSUE THREAD

**Issue by spadgos** [eslint/eslint#5587]
**What version of ESLint are you using?**: 2.4.0

**The problem I want to solve.**

`no-extra-parens` complains when using parens to disambiguate conditions using mixed boolean logic.

```js
if (x || (y && z)) { .. }
```

At my workplace, we have it as a recommendation to always use parentheses when mixing `&&` and `||`, since it can lead to confusion about the order of operations.

```js
// For example, 
x || y && z
// is that the same as
(x || y) && z
// or
x || (y && z)
```

Technically, in one of those last two lines of code, the parens are 'extra' and `no-extra-parens` complains, however I don't want to have to know which, nor do I want that being a source of bugs in my code if someone gets it wrong.

**Your take on the correct solution to problem.**
`no-extra-parens` could allow parens when being used to separate boolean operators. It might be nice to have it enforce using parens when mixing them, but that's probably now a different rule. As an example of what I mean:

```js
// OK
x = a || b || c;
x = a && b && c;
x = a && (b || c);
x = a && (b || c) && d;

// NOT OK
x = a || b && c;
```

**Comment by BYK** [comment]
This looks like a duplicate of #3065 so closing. Feel free to reopen if you disagree.

**Comment by spadgos** [comment]
You're right. Thanks @BYK.
